core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/tasks_shared.go

```diff
@@ -50,14 +50,24 @@ const (
 	runFilesArtifactsKey = "files_artifacts"
 
 	shellWrapperCommand = "/bin/sh"
+	taskLogFilePath     = "/tmp/kurtosis-task.log"
 	noNameSet           = ""
 	uniqueNameGenErrStr = "error occurred while generating unique name for the file artifact"
 
 	//  enables init mode on containers; cleaning up any zombie processes
 	tiniEnabled = true
 )
 
-var runTailCommandToPreventContainerToStopOnCreating = []string{"tail", "-f", "/dev/null"}
+var runCommandToStreamTaskLogs = []string{shellWrapperCommand, "-c", fmt.Sprintf("touch %s && tail -F %s", taskLogFilePath, taskLogFilePath)}
+
+// Wraps [commandToRun] to enable streaming logs from tasks.
+// Uses curly braces to execute the command(s) in the current shell.
+// Adds an extra echo to ensure each log ends with a newline.
+// Uses tee to direct output to the task log file while maintaining output to stdout.
+// Redirects stderr to stdout.
+func getCommandToRunForStreamingLogs(commandToRun string) []string {
+	return []string{shellWrapperCommand, "-c", fmt.Sprintf("{ %v; } %v %v %v %v %v %v %v %v", commandToRun, "2>&1", "|", "tee", taskLogFilePath, "&&", "echo", ">>", taskLogFilePath)}
```

Comment:
The extra echo adds an empty log line, but it's necessary because some tasks like read-prague-time in ethereum-package output without a newline, causing tail in the entrypoint to miss them. tail needs a newline to detect the log it seems. Let me know if there's a workaround; otherwise, we'll keep it until there's a complaint about the extra empty log line in tasks.